app/controllers/ships_controller.rb

```diff
@@ -50,77 +50,86 @@ def info
 
   def eject_cargo
     if params[:loader] && params[:amount] && current_user.can_be_attacked
-      amount = params[:amount].to_i rescue nil
+      amount = params[:amount].to_i
 
       if amount && (amount > 0)
         # check amount
-        render(json: { error_message: I18n.t('errors.you_dont_have_enough_of_this') }, status: 400) && (return) if Item.find_by(loader: params[:loader], spaceship: current_user.active_spaceship, equipped: false).count < amount
+        render(json: { error_message: I18n.t('errors.you_dont_have_enough_of_this') }, status: :bad_request) && (return) if Item.find_by(loader: params[:loader], spaceship: current_user.active_spaceship, equipped: false).count < amount
 
         EjectCargoWorker.perform_async(current_user.id, params[:loader], amount)
-        render(json: {}, status: 200) && (return)
+        render(json: {}, status: :ok) && (return)
       else
-        render(json: { error_message: I18n.t('errors.invalid_amount') }, status: 400) && (return)
+        render(json: { error_message: I18n.t('errors.invalid_amount') }, status: :bad_request) && (return)
       end
     end
-    render json: {}, status: 400
+    render json: {}, status: :bad_request
   end
 
   def insure
     if params[:id]
-      ship = Spaceship.find(params[:id]) rescue nil
-      if ship && (ship.user == current_user) && !ship.insured && current_user.docked
+      ship = Spaceship.ensure(params[:id])
+      if ship && (ship.user == current_user) && !ship.insured? && current_user.docked?
         price = (Spaceship.get_attribute(ship.name, :price) / 2).round
 
-        # check credits
-        render(json: { 'error_message': I18n.t('errors.you_dont_have_enough_credits') }, status: 400) && (return) unless current_user.units >= price
+        # check that they have enough units
+        if price > current_user.units
+          render(json: { 'error_message': I18n.t('errors.you_dont_have_enough_credits') }, status: :bad_request)
+          return
+        end
 
-        # Insure
-        ship.update_columns(insured: true)
+        # Insure & charge them
+        ActiveRecord::Base.transaction do
+          ship.update_columns(insured: true)
+          current_user.reduce_units(price)
+        end
 
-        # Deduct units
-        current_user.reduce_units(price)
-
-        render(json: {}, status: 200) && (return)
+        render(json: {}, status: :ok) && (return)
```

Comment:
Mabye get clear on if we should use `and` or `&&` and then change every other version to keep it consistent?

Reply:
Ruby Style Guide says "The and, or, and not keywords are banned. It's just not worth it. Always use &&, ||, and ! instead."
 and there's this: https://www.tinfoilsecurity.com/blog/ruby-demystified-and-vs